torch/utils/cpp_extension.py

```diff
@@ -489,19 +508,33 @@ def _check_cuda_version(compiler_name: str, compiler_version: TorchVersion) -> N
             )
 
 
+def _append_sycl_std_if_no_std_present(cflags):
+    if not any(flag.startswith('-sycl-std=') for flag in cflags):
+        cflags.append('-sycl-std=2020')
+
+
+def _wrap_sycl_host_flags(cflags):
+    host_cxx = get_cxx_compiler()
+    host_cflags = [
+        f'-fsycl-host-compiler={host_cxx}',
+        shlex.quote(f'-fsycl-host-compiler-options={cflags}'),
+    ]
+    return host_cflags
+
+
 class BuildExtension(build_ext):
     """
     A custom :mod:`setuptools` build extension .
 
     This :class:`setuptools.build_ext` subclass takes care of passing the
     minimum required compiler flags (e.g. ``-std=c++17``) as well as mixed
-    C++/CUDA compilation (and support for CUDA files in general).
+    C++/CUDA/SYCL compilation (and support for CUDA/SYCL files in general).
 
     When using :class:`BuildExtension`, it is allowed to supply a dictionary
     for ``extra_compile_args`` (rather than the usual list) that maps from
-    languages (``cxx`` or ``nvcc``) to a list of additional compiler flags to
```

Comment:
why they called nvcc is a language? Is it correct?

Reply:
It should be a typo.

Reply:
Note that this ended up in the API (see `nvcc` key below):
https://github.com/pytorch/pytorch/blob/379b54603a41f31e4524e83321497af255e50aeb/test/cpp_extensions/setup.py#L52